### PR TITLE
Modify tasks to use systemd or freshclam binary

### PR DESCRIPTION
**Pull request by ClementJ35**
## 🗣 Description ##

As describe in  #38, setup based on OSes doesn't make sense now as many system use systemd.
It's better to choose using the fact : ansible_service_mgr

## 🧪 Testing ##

Tested on Debian 11 & CentOS 8

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by ClementJ35** [comment]
Failing test seems imputable to docker instance : 
'''2022-01-31T17:01:49.4122921Z fatal: [fedora32]: FAILED! => {"changed": false, "msg": "failure 1 during daemon-reload: System has not been booted with systemd as init system (PID 1). Can't operate.\nFailed to connect to bus: Host is down\n"}'''
In the real world, Fedora 32 is systemd based.
Either ansible facts are wrong about systemd or the docker image have some ghosts information.

File: tasks/main.yml [hunk 1]
```diff
@@ -35,20 +35,13 @@
...
-        - setup_{{ ansible_distribution }}.yml
-        - setup_{{ ansible_os_family }}.yml
-      paths:
-        - "{{ role_path }}/tasks"
+- name: Load tasks file for systemd setup
+  ansible.builtin.include_tasks: setup_systemd.yml
+  when: ansible_service_mgr == 'systemd'
+
+- name: Load tasks file for bin setup
+  ansible.builtin.include_tasks: setup_bin.yml
```
Comment (jsf9k) [inline tasks/main.yml]: I think we should consider renaming `setup_bin.yml` to `setup_manual.yml`.  The latter more clearly indicates the purpose of the file (IMO).
```suggestion
- name: Load tasks file for manual setup
  ansible.builtin.include_tasks: setup_manual.yml
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
Thanks for another great improvement, @ClementJ35!  Please consider my change request while I fix the failing builds.

File: tasks/main.yml (hunk 1 above)
Comment (dav3r) [reply tasks/main.yml]: We also need to add some sort of documentation about this file and what it should contain.

**Comment by jsf9k** [comment]
> Failing test seems imputable to docker instance : '''2022-01-31T17:01:49.4122921Z fatal: [fedora32]: FAILED! => {"changed": false, "msg": "failure 1 during daemon-reload: System has not been booted with systemd as init system (PID 1). Can't operate.\nFailed to connect to bus: Host is down\n"}''' In the real world, Fedora 32 is systemd based. Either ansible facts are wrong about systemd or the docker image have some ghosts information.

I think Fedora 32 lacks the ClamAV SystemD service that is being manipulated in this role, even though it does run SystemD.

File: tasks/main.yml (hunk 1 above)
Comment (ClementJ35) [reply tasks/main.yml]: I did the changes mentioned above in [e485997](https://github.com/cisagov/ansible-role-clamav/pull/39/commits/e4859970e9b3a4cbaa125b75501620b9e038f4ac).
Can you check it's all right ?

Comment (jsf9k) [reply tasks/main.yml]: Looks good to me.  Thanks for making the changes.  @dav3r and @mcdonnnj, can you guys take another look?

**Review by jsf9k — APPROVED** [review]
Thanks for simplifying this Ansible role!

**Review by dav3r — APPROVED** [review]
This looks great to me now! 👍 👍

File: tasks/main.yml (hunk 1 above)
Comment (dav3r) [reply tasks/main.yml]: I'm satisfied with the recent changes for this - thanks.

Comment (mcdonnnj) [reply tasks/main.yml]: The updates work for me as well.

**Review by mcdonnnj — APPROVED** [review]
Strong work! 💪💪💪

**Pull request merged**